Makefile

```diff
@@ -1,26 +1,42 @@
+$(MAKE)FILES = $(shell find . -maxdepth 3 -type f -name $(MAKE)file)
+SUBDIRS   = $(filter-out ./,$(dir $($(MAKE)FILES)))
+ROOT_DIR = $(shell pwd)
+
 default: build
 
 all: build test
```

Comment:
Since `test` target internally calls `build` target, we can remove this redundant `build` from this `all` target here.